src/main/python/ayab/tests/test_ayab_control.py

```diff
@@ -31,157 +35,180 @@ def test__singlebed_2col(self):
-        assert ayab_control._circular_ribber( 7,3,9) == (1,0, 3,True ,False)
-        assert ayab_control._circular_ribber( 8,3,9) == (1,1, 4,False,False)
-        assert ayab_control._circular_ribber( 9,3,9) == (1,1, 4,True ,False)
-        assert ayab_control._circular_ribber(10,3,9) == (1,2, 5,False,False)
-        assert ayab_control._circular_ribber(11,3,9) == (1,2, 5,True ,False)
-        assert ayab_control._circular_ribber(12,3,9) == (2,0, 6,False,False)
-        assert ayab_control._circular_ribber(13,3,9) == (2,0, 6,True ,False)
-        assert ayab_control._circular_ribber(14,3,9) == (2,1, 7,False,False)
-        assert ayab_control._circular_ribber(15,3,9) == (2,1, 7,True ,False)
-        assert ayab_control._circular_ribber(16,3,9) == (2,2, 8,False,False)
-        assert ayab_control._circular_ribber(17,3,9) == (2,2, 8,True ,True )
+        assert ayab_control._circular_ribber( 0, 3) == (0, 0, 0, False, False)
+        assert ayab_control._circular_ribber( 1, 3) == (0, 0, 0, True , False)
+        assert ayab_control._circular_ribber( 2, 3) == (0, 1, 1, False, False)
+        assert ayab_control._circular_ribber( 3, 3) == (0, 1, 1, True , False)
+        assert ayab_control._circular_ribber( 4, 3) == (0, 2, 2, False, False)
+        assert ayab_control._circular_ribber( 5, 3) == (0, 2, 2, True , False)
+        assert ayab_control._circular_ribber( 6, 3) == (1, 0, 3, False, False)
+        assert ayab_control._circular_ribber( 7, 3) == (1, 0, 3, True , False)
+        assert ayab_control._circular_ribber( 8, 3) == (1, 1, 4, False, False)
+        assert ayab_control._circular_ribber( 9, 3) == (1, 1, 4, True , False)
+        assert ayab_control._circular_ribber(10, 3) == (1, 2, 5, False, False)
+        assert ayab_control._circular_ribber(11, 3) == (1, 2, 5, True , False)
+        assert ayab_control._circular_ribber(12, 3) == (2, 0, 6, False, False)
+        assert ayab_control._circular_ribber(13, 3) == (2, 0, 6, True , False)
+        assert ayab_control._circular_ribber(14, 3) == (2, 1, 7, False, False)
+        assert ayab_control._circular_ribber(15, 3) == (2, 1, 7, True , False)
+        assert ayab_control._circular_ribber(16, 3) == (2, 2, 8, False, False)
+        assert ayab_control._circular_ribber(17, 3) == (2, 2, 8, True , True )
         ayab_control._infRepeat = True
-        assert ayab_control._circular_ribber(18,3,9) == (0,0, 0,False,False)
+        assert ayab_control._circular_ribber(18, 3) == (0, 0, 0, False, False)
         ayab_control._startLine = 1
-        assert ayab_control._circular_ribber(12,3,9) == (0,0, 0,False,False)
+        assert ayab_control._circular_ribber(12, 3) == (0, 0, 0, False, False)
+
+    def test__select_needles(self):
+        ayab_control = AYABControl()
+        ayab_control._numColors = 3
+        ayab_control._startLine = 0
+        ayab_control._knitting_mode = 0
+        ayab_control._image = ayabImage(Image.new('P', (40,1)), 2)
```

Comment:
I would love to have short comments describing the test cases here, as I think it is not too obvious what is happening in the following lines.